### DISCUSSION TITLE
[bridge] how to install websockets in /server/middleware?

### DISCUSSION THREAD

**Discussion by HendrikJan** [nuxt/nuxt#16925] [Questions]
We currently have an Express server with Nuxt as middleware:
```javascript
// Only the pieces of code that are usefull for this question
const https = require('https');
const express = require('express');
const socketIo = require('socket.io');
const { loadNuxt, build } = require('nuxt');
const certificate = ....
const port = ....
const isDev = ....
const app = express();

app.use('/some/path/rendered/by/express', ....);

const nuxt = await loadNuxt(isDev ? 'dev' : 'start');
app.use(nuxt.render);
if (isDev) {
  build(nuxt);
}

var server = https.createServer(certificate, app).listen(port, async function () {
    const io = socketIo(server);
});
```

Now, we try to use @nuxt/bridge by running Nuxt and having our Express server as middleware:
```javascript
// /server/middleware/expressApp.js
import express from 'express';
import * as socketIo from 'socket.io';

const app = express();
app.use('/some/path/rendered/by/express', ....);

const io = new socketIo.Server();
io.attach(server); // ???? <-- how do we get "server" to attach to socket-io???

export default app;
```

As you can see, socket-io expects a `server`-object to attach to, but inside /server/middleware we have no server to attach to.
Doe anybody know how to solve this?

**Comment by lewebsimple** [comment]
Until the `listen` hook is made available in production or we have some way to customize the new node nitro preset I'm afraid this is only possible in development mode.
There are also rumors of a builtin WebSocket and/or SSE feature but I'm not sure it is a priority right now.

Maybe @pi0 or @danielroe can weigh in on this ?

**Comment by LeeGrobler** [comment]
Any news on this?
